uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/AbsenceTypeTest.php

```diff
@@ -681,4 +687,80 @@ public function testCalculateToilExpiryDateWhenAbsenceTypeAllowsAccrualsRequestA
     $expiry = $absenceType2->calculateToilExpiryDate($date);
     $this->assertEquals('2017-09-10', $expiry->format('Y-m-d'));
   }
+
+  public function testNonExpiredToilRequestsAreDeletedWhenToilIsDisabledForAbsenceType() {
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('-1 days'),
+      'end_date' => CRM_Utils_Date::processDate('+ 300days'),
+    ]);
+
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'title' => 'Title 1',
+      'allow_accruals_request' => true,
+      'max_leave_accrual' => 1,
+      'is_active' => 1,
+    ]);
+
+    TOILRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('+3 days'),
+      'to_date' => CRM_Utils_Date::processDate('+3 days'),
+      'to_date_type' => 1,
```

Comment:
no need for date types here and on all the other calls to TOILRequestFabricator and TOILRequest::validateParams